src/test/javascript/spec/component/lecture/pdf-preview.component.spec.ts

```diff
@@ -0,0 +1,433 @@
+                        getViewport: jest.fn(() => ({ width: 600, height: 800, scale: 1 })),
+                        render: jest.fn(() => ({
+                            promise: Promise.resolve(),
+                        })),
+                    }),
+                ),
+            }),
+        })),
+    };
+});
+
+jest.mock('pdfjs-dist/build/pdf.worker', () => {
+    return {};
+});
+
+function createMockEvent(target: Element, eventType = 'click'): MouseEvent {
+    const event = new MouseEvent(eventType, {
+        view: window,
+        bubbles: true,
+        cancelable: true,
+    });
+    Object.defineProperty(event, 'target', { value: target, writable: false });
+    return event;
+}
+
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ActivatedRoute } from '@angular/router';
+import { of, throwError } from 'rxjs';
+import { AttachmentService } from 'app/lecture/attachment.service';
+import { AttachmentUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/attachmentUnit.service';
+import { PdfPreviewComponent } from 'app/lecture/pdf-preview/pdf-preview.component';
+import { ElementRef } from '@angular/core';
+import { AlertService } from 'app/core/util/alert.service';
+import { HttpErrorResponse } from '@angular/common/http';
+
+describe('PdfPreviewComponent', () => {
+    let component: PdfPreviewComponent;
+    let fixture: ComponentFixture<PdfPreviewComponent>;
+    let attachmentServiceMock: any;
+    let attachmentUnitServiceMock: any;
+    let alertServiceMock: any;
+    let routeMock: any;
+    let mockCanvasElement: HTMLCanvasElement;
+    let mockEnlargedCanvas: HTMLCanvasElement;
+    let mockContext: any;
+    let mockOverlay: HTMLDivElement;
+
+    beforeEach(async () => {
+        global.URL.createObjectURL = jest.fn().mockReturnValue('mocked_blob_url');
+        attachmentServiceMock = {
+            getAttachmentFile: jest.fn().mockReturnValue(of(new Blob([''], { type: 'application/pdf' }))),
+        };
+        attachmentUnitServiceMock = {
+            getAttachmentFile: jest.fn().mockReturnValue(of(new Blob([''], { type: 'application/pdf' }))),
+        };
+        routeMock = {
+            data: of({
+                course: { id: 1, name: 'Example Course' },
+                attachment: { id: 1, name: 'Example PDF' },
+                attachmentUnit: { id: 1, name: 'Chapter 1' },
+            }),
+        };
+        alertServiceMock = {
+            addAlert: jest.fn(),
+            error: jest.fn(),
+        };
+
+        await TestBed.configureTestingModule({
+            imports: [PdfPreviewComponent],
+            providers: [
+                { provide: ActivatedRoute, useValue: routeMock },
+                { provide: AttachmentService, useValue: attachmentServiceMock },
+                { provide: AttachmentUnitService, useValue: attachmentUnitServiceMock },
+                { provide: AlertService, useValue: alertServiceMock },
```

Comment:
I think you'll have to add a provider for the translate service here (please double-check locally if the following snippet works).

```suggestion
            providers: [
                { provide: ActivatedRoute, useValue: routeMock },
                { provide: AttachmentService, useValue: attachmentServiceMock },
                { provide: AttachmentUnitService, useValue: attachmentUnitServiceMock },
                { provide: AlertService, useValue: alertServiceMock },
                { provide: TranslateService, useClass: MockTranslateService },
```


----

Test output:

```
    NullInjectorError: R3InjectorError(Standalone[PdfPreviewComponent])[_TranslateService -> _TranslateService -> _TranslateService]: 
      NullInjectorError: No provider for _TranslateService!

      at NullInjector.get (node_modules/@angular/core/fesm2022/core.mjs:1659:27)
      at R3Injector.get (node_modules/@angular/core/fesm2022/core.mjs:3105:33)
      at R3Injector.get (node_modules/@angular/core/fesm2022/core.mjs:3105:33)
      at R3Injector.get (node_modules/@angular/core/fesm2022/core.mjs:3105:33)
      at ChainedInjector.get (node_modules/@angular/core/fesm2022/core.mjs:5447:36)
      at lookupTokenUsingModuleInjector (node_modules/@angular/core/fesm2022/core.mjs:5800:39)
      at getOrCreateInjectable (node_modules/@angular/core/fesm2022/core.mjs:5848:12)
      at ɵɵdirectiveInject (node_modules/@angular/core/fesm2022/core.mjs:11936:19)
      at NodeInjectorFactory.TranslateDirective_Factory [as factory] (ng:/TranslateDirective/ɵfac.js:6:7)
      at getNodeInjectable (node_modules/@angular/core/fesm2022/core.mjs:6060:44)
      at instantiateAllDirectives (node_modules/@angular/core/fesm2022/core.mjs:12806:27)
      at createDirectivesInstances (node_modules/@angular/core/fesm2022/core.mjs:12202:5)
      at ɵɵelementStart (node_modules/@angular/core/fesm2022/core.mjs:24362:9)
      at ɵɵelement (node_modules/@angular/core/fesm2022/core.mjs:24420:5)
      at PdfPreviewComponent_Template (ng:/PdfPreviewComponent.js:109:9)
      at executeTemplate (node_modules/@angular/core/fesm2022/core.mjs:12162:9)
      at renderView (node_modules/@angular/core/fesm2022/core.mjs:13369:13)
      at renderComponent (node_modules/@angular/core/fesm2022/core.mjs:13315:5)
      at renderChildComponents (node_modules/@angular/core/fesm2022/core.mjs:13417:9)
      at renderView (node_modules/@angular/core/fesm2022/core.mjs:13397:13)
      at ComponentFactory.create (node_modules/@angular/core/fesm2022/core.mjs:17089:17)
      at initComponent (node_modules/@angular/core/fesm2022/testing.mjs:2039:51)
      at _ZoneDelegate.invoke (node_modules/zone.js/bundles/zone.umd.js:416:32)
      at Object.onInvoke (node_modules/@angular/core/fesm2022/core.mjs:7241:33)
      at _ZoneDelegate.invoke (node_modules/zone.js/bundles/zone.umd.js:415:38)
      at ZoneImpl.run (node_modules/zone.js/bundles/zone.umd.js:147:47)
      at NgZone.run (node_modules/@angular/core/fesm2022/core.mjs:7087:28)
      at _TestBedImpl.createComponent (node_modules/@angular/core/fesm2022/testing.mjs:2051:41)
      at Function.createComponent (node_modules/@angular/core/fesm2022/testing.mjs:1842:37)
      at src/test/javascript/spec/component/lecture/pdf-preview.component.spec.ts:89:27
```